File: scsc-monolith/src/main/java/demo/scsc/api/payment/ProcessPaymentCommand.java
package demo.scsc.api.payment;

import org.axonframework.modelling.command.TargetAggregateIdentifier;

import java.math.BigDecimal;
import java.util.UUID;

public record ProcessPaymentCommand(
        @TargetAggregateIdentifier UUID orderPaymentId,
        BigDecimal amount
) {

}

File: scsc-monolith/src/main/kotlin/demo/scsc/api/payment/ProcessPaymentCommand.kt
package demo.scsc.api.payment

import org.axonframework.modelling.command.TargetAggregateIdentifier
import java.math.BigDecimal
import java.util.*

data class ProcessPaymentCommand(
    @TargetAggregateIdentifier val orderPaymentId: UUID,
    val amount: BigDecimal
)
